### PR TITLE
Update CONTRIBUTING.md with more Review App info

### PR DESCRIPTION
**Pull request by damassi**
Updates review app creation steps based on pairing session with @dleve123

### CODE REVIEW

**Comment by damassi** [comment]
@dleve123 - Some notes from the output:

1) It doesn't look like it outputted the fully resolved external endpoint
a193ef1e6e63411e9aff2127c3b176f7-543241782.us-east-1.elb.amazonaws.com

```sh
xperimental-app-shell: digest: sha256:caff3e049b7c8984d13bf5457abb738bb614c960b87dcd2b06b9a47a23e10afa size: 2425
Pushed hokusai_force:latest to 585031190124.dkr.ecr.us-east-1.amazonaws.com/force:experimental-app-shell


# Edit the K8S YAML to reference the proper Docker image
# sed -i '' 's/:staging/:auct-281/g' hokusai/auct-281.yml
#
# The weird `-i ''` is needed on MacOS:
# https://stackoverflow.com/a/44864004
# WARNING: This might not work on Linux.
sed -i '' "s/:staging/:$NAME/g" hokusai/$NAME.yml

# Edit the K8S YAML to remove the instructions that enforce that the service
# can only be accessible via Cloudflare
#
# First, remove the `loadBalancerSourceRanges:` line
sed -i '' '/loadBalancer/d' hokusai/$NAME.yml

# Then, delete all the IP address lines.
#
# WARNING: This is a bit sketchy as this will
# delete any line of the form "- [number][number][number].". This is my best
# approx for an regex for an IP address, and I'm sure that there are better
# ones.
sed -i '' "/- [[:digit:]][[:digit:]][[:digit:]]./d" hokusai/$NAME.yml

# Provision the review app
hokusai review_app create $NAME --verbose
namespace "experimental-app-shell" created
deployment.extensions "force-web" created
horizontalpodautoscaler.autoscaling "force-web" created
service "force-web" created
Created Kubernetes environment /Users/cn/Sites/artsy/force/hokusai/experimental-app-shell.yml

# Copy Force staging's ConfigMap to your review app
hokusai review_app env copy $NAME --verbose
==> hokusai exec `kubectl --context staging get configmap force-environment -o yaml`

==> hokusai exec `kubectl --context staging --namespace experimental-app-shell apply -f /var/folders/5f/kd05sp0n0dd1zmpkkh3mmtmw0000gn/T/tmpohBzXY`


# Copy the staging-shared nginx config to your review app
==> hokusai exec `kubectl --context staging --namespace default get configmap nginx-config -o yaml`

==> hokusai exec `kubectl --context staging --namespace experimental-app-shell apply -f /var/folders/5f/kd05sp0n0dd1zmpkkh3mmtmw0000gn/T/tmp4ovDL9`

2) In order to find the url, I had to run 
```sh
hokusai review_app status experimental-app-shell
```
where I could then grab the external IP:
```
a193ef1e6e63411e9aff2127c3b176f7-543241782.us-east-1.elb.amazonaws.com
```
However, going to this, it then redirects to `staging.artsy.net`, without me adding anything to DynDNS. 

I've just updated Dyn with this url, lets see what happens...

**Comment by joeyAghion** [comment]
@damassi when you try to access the load-balancer directly, it's probably redirecting you to what it thinks is its SSL URL (but it's not, since the review app's `APP_URL` was probably just copied from staging).

For SSL to work correctly on the review app, you'll need to access it at a *.artsy.net name. This can be done with an actual DNS entry, or a trick that works locally is to add a line to the `/etc/hosts` file that associates a valid name to an IP address for that load-balancer. E.g.:

```
dig a193ef1e6e63411e9aff2127c3b176f7-543241782.us-east-1.elb.amazonaws.com
...
;; ANSWER SECTION:
... 3.227.87.243
```
Then add to `/etc/hosts` a line like:

```
myreviewapp.artsy.net 3.227.87.243
```

**Comment by damassi** [comment]
@joeyAghion - I've created a Dyn entry pointing to

http://experimental-app-shell.artsy.net

But it redirects to `staging.artsy.net`. Perhaps I have to update APP_URL to `https://experimental-app-shell.artsy.net`?

**Comment by joeyAghion** [comment]
Yes I think you have to set `APP_URL` to the correct URL. Did you `hokusai review_app refresh <name>` after setting the updated config?

This is the line that redirects to `APP_URL`: https://github.com/artsy/force/blob/84446b89032ce2c5b01d182cf6a0aa080c2dcb85/src/lib/middleware/ensure_ssl.coffee#L11

I can successfully hit https://experimental-app-shell.artsy.net, but I see those asset `403` failures. I'm not familiar enough with the asset compilation to understand if that's due to the image being hosted or the config somehow.

**Comment by dleve123** [comment]
Notably the script does set [`APP_URL`](https://github.com/artsy/force/blob/master/scripts/build_review_app.sh#L75), but there isn't a step that forces getting new pods via restart. I wonder if the 2nd to last step [(after the env-setting step)](https://github.com/artsy/force/blob/master/scripts/build_review_app.sh#L74-L77) should be a `hokusai review_app deploy $name`. This should kill 2 bids with 1 stone by:

1. Ensuring that we get fresh pods, with the updated environment variables
2. Ensuring that we run `yarn publish-assets` – ensuring that we are building and using the latest client.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -69,13 +69,30 @@ This creates a production-ready bundle of client and server-side code and boots
...
+
+If you'd like a pretty URL subdomain or need to test full OAuth flows (for, say, login redirects between Gravity and Force for Auction registration) then an additional non-automated step is required via DynDNS:
+
+1) [Login to DynDNS](https://manage.dynect.net/qcke/artsy.net) 
+1) Click `Add New Record`
+1) Enter a new subdomain under `Node`
+1) Change `Type` dropdown to `CNAME`
+1) Paste in URL output by `build_review_app.sh` script into `RData` field, and add a `.` at the end
+1) Scroll to bottom and hit `Save` and then publish changes
+1) DNS will propagate and after a few minutes the review app will be available via `<your-supdomain>.artsy.net`
```
Comment (dleve123) [inline CONTRIBUTING.md]:
```suggestion
1) DNS will propagate and after a few minutes the review app will be available via `<your-subdomain>.artsy.net`
```

**Comment by damassi** [comment]
@dleve123 - see https://github.com/artsy/force/pull/4629

**Pull request merged**